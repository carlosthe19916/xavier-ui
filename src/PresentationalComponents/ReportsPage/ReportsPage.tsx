import React, { Fragment, Component } from 'react';

import {
    Main,
    PageHeader,
    PageHeaderTitle
} from '@redhat-cloud-services/frontend-components';

interface Props {
    mainStyle?: any;
};

interface State {
}

class ReportsPage extends Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    public render() {
        const { children } = this.props;

        return (
            <Fragment>
                <PageHeader>
                    <PageHeaderTitle title={ 'Reports' } />
                </PageHeader>
                <Main style={ this.props.mainStyle }>
                    { children }
                </Main>
            </Fragment>
        );
    }
};

export default ReportsPage;
