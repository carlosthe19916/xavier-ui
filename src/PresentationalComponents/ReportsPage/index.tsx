import ReportsPage from './ReportsPage';
import { withRouter } from 'react-router';

export default withRouter(ReportsPage);
