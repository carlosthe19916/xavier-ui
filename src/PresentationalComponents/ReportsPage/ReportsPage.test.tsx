import { shallow } from "enzyme";
import ReportsPage from "./ReportsPage";

describe("ReportListPage", () => {
  const routerProps = {
    history: null,
    location: null,
    match: null
  };

  it("expect to render without breadcrumb", () => {
    const wrapper = shallow(
      <ReportsPage title="My title" {...routerProps}>
        children
      </ReportsPage>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render with breadcrumb and without showing rootLink", () => {
    const props = Object.assign({}, routerProps, {
      history: {
        location: {
          pathname: "/reports"
        }
      }
    });

    const wrapper = shallow(
      <ReportsPage title="My title" showBreadcrumb={true} {...props}>
        children
      </ReportsPage>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render with breadcrumb and showing rootLink", () => {
    const props = Object.assign({}, routerProps, {
      history: {
        location: {
          pathname: "/mypath"
        }
      }
    });

    const wrapper = shallow(
      <ReportsPage title="My title" showBreadcrumb={true} {...props}>
        children
      </ReportsPage>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
