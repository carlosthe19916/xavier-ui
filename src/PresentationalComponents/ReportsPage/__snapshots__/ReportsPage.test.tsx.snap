// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportPage expect to render 1`] = `
<Fragment>
  <PageHeader>
    <PageHeaderTitle
      title="Reports"
    />
  </PageHeader>
  <Connect(Main)
    style={
      Object {
        "color": "red",
      }
    }
  >
    children
  </Connect(Main)>
</Fragment>
`;
